### PR TITLE
Made some methods protected to allow subclassing of RedisClient

### PR DESCRIPTION
**Pull request by jruaux**
Make sure that:

- [ ] You have read the [contribution guidelines](https://github.com/lettuce-io/lettuce-core/blob/main/.github/CONTRIBUTING.md).
- [ ] You have created a feature request first to discuss your contribution intent. Please reference the feature request ticket number in the pull request.
- [ ] You use the code formatters provided [here](https://github.com/lettuce-io/lettuce-core/blob/main/formatting.xml) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.

### CODE REVIEW

**Comment by jruaux** [comment]
Feature request #1490

File: src/main/java/io/lettuce/core/RedisClient.java [hunk 1]
```diff
@@ -805,15 +805,15 @@ private static void checkValidRedisURI(RedisURI redisURI) {
         }
     }
 
-    private static <K, V> void assertNotNull(RedisCodec<K, V> codec) {
+    protected static <K, V> void assertNotNull(RedisCodec<K, V> codec) {
```
Comment (dengliming) [inline src/main/java/io/lettuce/core/RedisClient.java:808]: A little confused. Seems `LettuceAssert` already solved what you need.

Comment (jruaux) [reply src/main/java/io/lettuce/core/RedisClient.java:808]: The `assertNotNull` methods are not super critical. I can take them out of the PR if needed.

Comment (mp911de) [reply src/main/java/io/lettuce/core/RedisClient.java:808]: Please revert that change and use `LettuceAssert.notNull(…)` instead.

**Review by mp911de — COMMENTED** [review]
I'm not fully sure that's the right path. Can you elaborate a bit why you can't call `connectAsync` directly? That seems the logical choice since the method calls `connectStandaloneAsync` and `transformAsyncConnectionException`.

**Comment by jruaux** [comment]
You're correct, I can use connectAsync directly. I need to get more sleep :man_facepalming:
Please ignore this issue and PR. Sorry for wasting your time!

**Comment by mp911de** [comment]
No worries, understanding the requirements and what's available is always worth a discussion. Before submitting a PR, please file an issue the next time so we can discuss any requirements there.

**Pull request closed without merging**